andela/magma-backend
# #167749951 Signup credentials Validation

#### What does this PR do?

This PR validates credentials for user signup endpoint

#### Description of Tasks to be completed

- Write tests for user signup validations
- Install `@hapi/joi` to validate user credentials
- Install `lodash` to make use of its utility functions
- Setup user signup schema
- Validate with `joi`
- Test with postman
- Refactor test to ensure all cases are passing
- Update Swagger documentation

#### How should this be manually tested?

- Clone the Repo, change into the directory and run `npm install`
- Run `npm test` to see all test cases
- Run `npm dev` to start the server
- Make a `POST` request to `http://localhost:{PORT}/api/v1/users/signup` on Postman using invalid credentials

Sample request body
```
{
    "email": "naimatdavidmail.com",
    "firstname": "Ibrahim99",
    "lastname": "Jibson6",
     "password": "chibuokemalabi"
}
```


#### Relevant pivotal tracker story

[167749951](https://www.pivotaltracker.com/story/show/167749951)


#### Screenshots

![Screenshot (124)](https://user-images.githubusercontent.com/43535158/64074193-322aa500-cca0-11e9-9046-d5481925e896.png)

![Screenshot (125)](https://user-images.githubusercontent.com/43535158/64074202-3ce53a00-cca0-11e9-954e-d517ad8914e4.png)
